### PR TITLE
Implemented tests for simulations and fixed bugs accordingly.

### PR DESCRIPTION
**Pull request by neveritt**
The tests have been implemented as fixtures, i.e. different
realizations of a common system is simulated and their
time responses are compared to the known analytic response.

### CODE REVIEW

File: src/methods/simulate.jl [hunk 1]
```diff
@@ -18,15 +18,16 @@ function simulate{T}(sys::LtiSystem{Val{T},Val{:cont}}, tspan;
 
   f     = (t,x,dx)->sys(t,x,dx,input)
   simvar= SimType(initial, zeros(numoutputs(sys)), zeros(numinputs(sys)))
+  f(tspan[1], simvar, zeros(initial))
```
Comment (aytekinar) [inline src/methods/simulate.jl]: wow.. do we need this? was it a bug?

Comment (aytekinar) [inline src/methods/simulate.jl:21]: wow.. do we need this? was it a bug?

File: src/methods/simulate.jl [hunk 2]
```diff
@@ -18,15 +18,16 @@ function simulate{T}(sys::LtiSystem{Val{T},Val{:cont}}, tspan;
...
   prob  = ODEProblem(f, simvar, tspan)
   sln   = OrdinaryDiffEq.solve(prob, alg; tstops = tstops, kwargs...)
 
   tvals = sln.t
-  xvals = Matrix{eltype(tvals)}(length(tvals), numstates(sys))
-  yvals = Matrix{eltype(tvals)}(length(tvals), numoutputs(sys))
-  uvals = Matrix{eltype(tvals)}(length(tvals), numinputs(sys))
+  xvals = Matrix{eltype(sln[1].x)}(length(tvals), numstates(sys))
+  yvals = Matrix{eltype(sln[1].y)}(length(tvals), numoutputs(sys))
+  uvals = Matrix{eltype(sln[1].u)}(length(tvals), numinputs(sys))
```
Comment (aytekinar) [inline src/methods/simulate.jl:30]: I think these were not needed --- DiffEq gives solutions and the time steps with the same data type. Yet, reading this is easier for the eye :)

File: src/methods/simulate.jl [hunk 3]
```diff
@@ -42,16 +43,19 @@ function simulate{T}(sys::LtiSystem{Val{T},Val{:disc}}, tspan;
...
   simvar= SimType(initial, zeros(numoutputs(sys)), zeros(numinputs(sys)))
+  temp = zeros(simvar.x)
+  f(0.0, simvar, temp)  # ensure first step is correct
+  simvar.x = temp 
 
-  prob  = DiscreteProblem(f, simvar, tspan)
   dt    = samplingtime(sys)
+  prob  = DiscreteProblem(f, simvar, tspan) # extra time step, see below
   sln   = OrdinaryDiffEq.solve(prob, FunctionMap(scale_by_time=false);
     kwargs..., dt = samplingtime(sys))
```
Comment (aytekinar) [inline src/methods/simulate.jl]: `dt = samplingtime(sys)` above is not used. we should remove it. or, we can write here as,
```julia
sln = OrdinaryDiffEq.solve(prob, FunctionMap(scale_by_time=false); kwargs..., dt = dt)
```

File: src/methods/simulate.jl [hunk 4]
```diff
@@ -42,16 +43,19 @@ function simulate{T}(sys::LtiSystem{Val{T},Val{:disc}}, tspan;
...
 
-  tvals = sln.t
-  xvals = Matrix{eltype(tvals)}(length(tvals), numstates(sys))
-  yvals = Matrix{eltype(tvals)}(length(tvals), numoutputs(sys))
-  uvals = Matrix{eltype(tvals)}(length(tvals), numinputs(sys))
+  tvals = sln.t[1:end]
```
Comment (aytekinar) [inline src/methods/simulate.jl]: why slicing? isn't `tvals = sln.t` same as `tvals = sln.t[1:end]`?

File: src/types/system/ltisystem.jl [hunk 5]
```diff
@@ -41,115 +41,115 @@ function numoutputs(s::LtiSystem)
...
-function ss(s::LtiSystem)
-  warn("ss(s) not implemented for s::$(typeof(s))")
-  throw(MethodError(ss, Tuple{typeof(s)}))
-end
-
-function tf(s::LtiSystem)
-  warn("tf(s) not implemented for s::$(typeof(s))")
-  throw(MethodError(tf, Tuple{typeof(s)}))
-end
+# # Iteration interface (meaningful in MIMO)
```
Comment (aytekinar) [inline src/types/system/ltisystem.jl:44]: should we simply remove them all?

File: src/types/system/statespace.jl [hunk 6]
```diff
@@ -53,7 +53,7 @@ immutable StateSpace{T,S,M1,M2,M3,M4} <: LtiSystem{T,S}
     end
     dx[:] = A*x + B*ucalc
   end
-  function (sys::StateSpace)(t::Real, x::DEDataArray, dx::AbstractVector, u)
+  function (sys::StateSpace)(t::Real, x::DiffEqBase.DEDataArray, dx::AbstractVector, u)
```
Comment (aytekinar) [inline src/types/system/statespace.jl:56]: wasn't it working without the namespace resolution, `DiffEqBase.`? Sorry... :( Maybe we should be `using DiffEqBase: DEDataArray`?

**Review by aytekinar — DISMISSED** [review]
I have checked the tests --- they really look cute :)

I just wanted to ask some questions, that's why I am "requesting changes". You can ignore the option and merge right away. I think there might be just two small glitches as in `dt` and `sln.t[1:end]` type of things. Or, I am missing something :)

Anyways... I will try to fix some tests following your approach. `@testset`s are really cool btw.

Final remark: I don't know why OSX tests fail. Any opinion?

File: src/methods/simulate.jl (hunk 2 above)
Comment (neveritt) [reply src/methods/simulate.jl:30]: This was needed for SysId, to be able to take the derivative of a cost function based on the simulation response of a parameterized LtiSystem.

**Pull request merged**